### ISSUE TITLE
first experiments

### ISSUE THREAD

**Issue by ArneBinder** [Cora4NLP/multi-task-knowledge-transfer#49]
# general

all the experiments should be executed with the multi-model variant of the individual task models
- [ ] target-model only
- [ ] **frozen** pre-trained target-model + bert-base-cased
- [ ] **frozen** pre-trained target-model + **frozen** other-task-model
- [ ] pre-trained target-model + **frozen** other-task-model

Hyperparameters:
 - learning rate
 - training time: start with a lot of epochs (50?); early stopping (see `patience` parameter)?
 - warming: for later (seems interesting, but requires effort)

# Co-ref
@tanikina @ArneBinder 
- [x] target-model only: 
   - train target: [config](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/configs/experiment/conll2012_coref_hoi_multimodel_train_target.yaml), [W&B run](https://wandb.ai/tanikina/conll2012-multi_model_coref_hoi-training/runs/miywcbq8), [results](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/log.md#coreference-resolution-target-only-model-with-attention)
   - frozen target: [config](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/configs/experiment/conll2012_coref_hoi_multimodel_frozen_target.yaml), [W&B run](https://wandb.ai/tanikina/conll2012-multi_model_coref_hoi-training/runs/7uyjay9c), [results](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/log.md#coreference-resolution-pre-trained-frozen-target-only-model-with-attention)
   - train target, but with **aggregate=mean(!)**: [W&B run](https://wandb.ai/tanikina/conll2012-multi_model_coref_hoi-training/runs/3dal5i30), [results](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/log.md#coreference-resolution-target-only-model)
- [x] **frozen** pre-trained target-model + bert-base-cased: [config](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/configs/experiment/conll2012_coref_hoi_multimodel_frozen_target_with_bert.yaml), [W&B run](https://wandb.ai/tanikina/conll2012-multi_model_coref_hoi-training/runs/ce4ye393), [results](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/log.md#coreference-resolution-frozen-target-pre-trained-coreference-model-and-bert-base-cased)
- [x] **frozen** pre-trained target-model + **frozen** other-task-model:
    - other=NER: [W&B run](https://wandb.ai/tanikina/conll2012-multi_model_coref_hoi-training/runs/jazixec9), [results](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/log.md#coreference-resolution-frozen-target-pre-trained-coreference-model-and-frozen-ner)
    - other=RE: [config](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/configs/experiment/conll2012_coref_hoi_multimodel_frozen_target_with_frozen_re.yaml), [W&B run](https://wandb.ai/tanikina/conll2012-multi_model_coref_hoi-training/runs/sa0bco64), [results](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/log.md#coreference-resolution-frozen-target-pre-trained-coreference-model-and-frozen-re-tacred)
    - other=EQA: [W&B run](https://wandb.ai/tanikina/conll2012-multi_model_coref_hoi-training/runs/3frydgsh), [results](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/log.md#coreference-resolution-frozen-target-pre-trained-coreference-model-and-frozen-squad)
- [x] pre-trained target-model + **frozen** other-task-model:
   - other=NER: [W&B run](https://wandb.ai/tanikina/conll2012-multi_model_coref_hoi-training/runs/dolw92mi), [results](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/log.md#coreference-resolution-tuned-target-pre-trained-coreference-model-and-frozen-ner)
   - other=RE: [config](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/configs/experiment/conll2012_coref_hoi_multimodel_tuned_target_with_frozen_re.yaml), [W&B run](https://wandb.ai/tanikina/conll2012-multi_model_coref_hoi-training/runs/lbzjb6z6), [results]()
   - other=EQA: [W&B run](https://wandb.ai/tanikina/conll2012-multi_model_coref_hoi-training/runs/5d1irxwl), [results](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/log.md#coreference-resolution-tuned-target-pre-trained-coreference-model-and-frozen-squad)

EDIT: 
 - PR that adds respective configs and updates the `log.md`: #64
 - experimental results: [log.md#2023-09-28](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/log.md#2023-09-28)
 - wandb report with the val/f1 and val/loss graphs (experiments from 2023-09-28 and 2023-09-29): https://wandb.ai/tanikina/conll2012-multi_model_coref_hoi-training/reports/Coreference-Experiments--Vmlldzo1NjAwNTMy

# NER
@harbecke 
- [x] target-model only
- [x] **frozen** pre-trained target-model + bert-base-cased
- [x] **frozen** pre-trained target-model + **frozen** other-task-model
- [x] pre-trained target-model + **frozen** other-task-model

PR that adds the NER configs and update the experiment log: #71 

# RE
@leonhardhennig 
- [x] target-model only: [config](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/configs/experiment/tacred_multimodel.yaml), [W&B run](https://wandb.ai/leonhardhennig/tacred-multi_model_re_text_classification-training/runs/5oo1xfcc), results(TODO @ArneBinder)
- [x] **frozen** pre-trained target-model + bert-base-cased: [config](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/configs/experiment/tacred_multimodel_frozen_re_tuned_bert.yaml), [W&B run](https://wandb.ai/leonhardhennig/tacred-multi_model_re_text_classification-training/runs/40dmu49m), results(TODO @ArneBinder)
- [x] **frozen** pre-trained target-model + **frozen** other-task-model: [config](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/configs/experiment/tacred_multimodel_frozen_re_ner.yaml) [W&B run](https://wandb.ai/leonhardhennig/tacred-multi_model_re_text_classification-training/runs/wlcdze53) [results](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/log.md#relation-extraction---frozen-pre-trained-target-model--frozen-ner-model-with-attention)
- [x] pre-trained target-model + **frozen** other-task-model: [config](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/configs/experiment/tacred_multimodel_re_frozen_ner.yaml) [W&B run](https://wandb.ai/leonhardhennig/tacred-multi_model_re_text_classification-training/runs/467lkbxr) [results](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/log.md#relation-extraction---pre-trained-target-model--frozen-ner-model-with-attention)

## Sanity Checks
- [x] **frozen**  pre-trained target-model with mean "aggregation" [config](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/configs/experiment/tacred_multimodel_frozen_re.yaml) + "model.aggregate=mean" [W&B run](https://wandb.ai/leonhardhennig/tacred-multi_model_re_text_classification-training/runs/o6pl2lxs) [results]()
- [x] **frozen**  pre-trained target-model with attention-based "aggregation" [config](https://github.com/Cora4NLP/multi-task-knowledge-transfer/blob/main/configs/experiment/tacred_multimodel_frozen_re.yaml) [W&B run](https://wandb.ai/leonhardhennig/tacred-multi_model_re_text_classification-training/runs/w0qx3s7u) [results]()

# extractive QA
@StalVars 
- [ ] target-model only
- [ ] **frozen** pre-trained target-model + bert-base-cased
- [ ] **frozen** pre-trained target-model + **frozen** other-task-model
- [ ] pre-trained target-model + **frozen** other-task-model

Weights & Biases project for EQA: https://wandb.ai/stalvars/dataset+squadv2-task+extractive_question_answering-training
